src/librustc_mir/build/scope.rs

```diff
@@ -141,6 +144,9 @@ pub struct Scope<'tcx> {
 
     /// The cache for drop chain on “normal” exit into a particular BasicBlock.
     cached_exits: FxHashMap<(BasicBlock, CodeExtent), BasicBlock>,
+
+    /// The cache for drop chain on "generator drop" exit.
+    cached_generator_drop: Option<BasicBlock>,
```

Comment:
Why do we have like 3 separate ways of caching scope cleanup? There should be exactly 1 way.